### PR TITLE
Correct name diagram filename

### LINKED ISSUE #118: Modify the name of the txt file for the diagrams when the name of the generator contains a "/"
In certain cases the name of a generator can contain a "/".
This leads to an error when loading the PQ diagram as the name of the txt file contains this "/".

Maybe we could replace it in the name of the txt file.

### PR DESCRIPTION
**Pull request by lecourtoisflo**
closes #118

### CODE REVIEW

File: scripts/updateMainReference.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
```
Comment (cossartque) [inline scripts/updateMainReference.py:1]: What is this file for ?

File: sources/Outputs/src/Constants.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
+//
+// Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (cossartque) [inline sources/Outputs/src/Constants.cpp:2]: 2021 ?

File: scripts/updateMainReference.py (hunk 1 above)
Comment (rosiereflo) [reply scripts/updateMainReference.py:1]: Please add this kind of utility file in a dedicated commit, with the associated doc and comments

File: sources/Outputs/src/Constants.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace dfl {
+namespace outputs {
+namespace constants {
+
+std::string
+diagramFilename(const dfl::algo::GeneratorDefinition& generator) {
+  // Remove '/' and '\' characters from id and replace it with '%'
+  auto id = generator.id;
+  std::replace(id.begin(), id.end(), '/', '%');
```
Comment (rosiereflo) [inline sources/Outputs/src/Constants.cpp:30]: % can be a special character in Windows if I'm not mistaken (%20 for space for example in html), so I would propose to use '-' or '_'

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: sources/Outputs/src/Constants.cpp (hunk 3 above)
Comment (marcochiaramello) [reply sources/Outputs/src/Constants.cpp:30]: I'd use '_'

File: sources/Outputs/src/Constants.cpp (hunk 2 above)
Comment (lecourtoisflo) [reply sources/Outputs/src/Constants.cpp:2]: OK

File: sources/Outputs/src/Constants.cpp (hunk 3 above)
Comment (lecourtoisflo) [reply sources/Outputs/src/Constants.cpp:30]: OK

File: scripts/updateMainReference.py (hunk 1 above)
Comment (rosiereflo) [reply scripts/updateMainReference.py:1]: I meant a dedicated PR, sorry

**Review by cossartque — APPROVED** [review]

**Review by rosiereflo — APPROVED** [review]

**Review by marcochiaramello — APPROVED** [review]

**Pull request merged**